INSERT INTO `pages` (`id`, `name`, `slug`, `seo_desc`, `seo_keywords`, `text`, `desc`, `seo_title`, `is_active`)
VALUES (1, 'Gene-Calc', '/',
        'Gene-Calc allows to calculate commonly used parameters, tests and formulas applied in biological science such as: chi-square, PIC and H, genetic balance and genetic distance.',
        'bioinformatic tool, polymorphic information content (PIC), heterozygosity (H), genetic distance, hardy-weinberg equilibrium, online calculator, chi-square tests',
        '<p><span style=\"font-size:18px\"><strong>The Gene-Calc is a constantly developed tool dedicated to calculations related to biological sciences, especially focused in the field of genetics. Application offers:</strong></span></p>\r\n\r\n<ul>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\">testing of compliance with the Hardy-Weinberg model</span></li>\r\n	<li><span style=\"font-size:16px\">tests from the chi-square family with coefficients of correlation, for contingency tables of any size</span></li>\r\n	<li><span style=\"font-size:16px\">determination of Polymorphic Information Content (PIC) and Heterozygosity (H) values based on both codominant and dominant markers</span></li>\r\n	<li><span style=\"font-size:16px\">estimation of&nbsp;genetic distance</span></li>\r\n	<li><span style=\"font-size:16px\">sequences&nbsp;tools (module provide a full connection with the NCBI database)</span></li>\r\n</ul>\r\n\r\n<p><span style=\"font-size:18px\"><strong>Upcoming upgrades:</strong></span></p>\r\n\r\n<ul>\r\n	<li><span style=\"font-size:16px\">Hardy-Weinberg calculator for multiallelic loci.</span></li>\r\n	<li><span style=\"font-size:16px\">Tool to examine similarity&nbsp;between sets of genes (all vs all), with heatmap and dendrogram [<a href=\"https://github.com/Janbinkowski96/bio_tools/tree/master/distance%20based%20off%20sequence\" target=\"_blank\">code of basic functional version available</a>].</span></li>\r\n	<li><span style=\"font-size:16px\">Tool to examine compliance&nbsp;with&nbsp;the&nbsp;Gaussian distribution [<a href=\"https://github.com/Janbinkowski96/stat_tools\" target=\"_blank\">code of basic functional version available</a>].</span></li>\r\n</ul>\r\n\r\n<hr />\r\n<p><span style=\"font-size:18px\"><strong>About project</strong></span></p>\r\n\r\n<p style=\"text-align:justify\"><span style=\"font-size:16px\">New techniques of lab work generate large result data, which are impossible to analyse without using bioinformatic solutions. Commercially available tools could be hard to use for beginners and what is the most important - expensive. On the other hand noncommercial apps often are imperfect. The aim of our work is to create a full free app with lack of restrictions.<br />\r\nOur team is created by students of polish universities of technology, keen on looking for solutions to scientific problems.&nbsp;</span></p>\r\n\r\n<hr />\r\n<p><span style=\"font-size:18px\"><strong>How to cite this page:</strong></span></p>\r\n\r\n<p><strong><span style=\"font-size:14px\">Bińkow</span><span style=\"font-size:14px\">ski J., Miks S., (2018, September). Gene-Calc [Computer software]. Available from: www.gene-calc.pl.</span></strong></p>\r\n\r\n<hr />',
        '<p style=\"text-align:center\"><a href=\"contact@gene-calc.pl\">contact@gene-calc.pl</a></p>\r\n',
        'Gene-Calc – Bioinformatic Tool', 1),
       (2, 'Hardy-Weinberg equilibrium', '/hardy-weinberg-page',
        'Tool to examine compliance between expected and observed number of genotypes.',
        'Hardy-Weinberg equilibrium, genetics balance, fixation index (fis), Hardy-Weinberg online calculator, Hardy-Weinberg calculator online',
        '<p style=\"text-align:justify\">This tool allows examining the compliance between observed and expected number of genotypes based on Hardy-Weinberg law. The users insert numbers of homozygous, heterozygous and rare homozygous individuals and the significance level - alfa. Output values are the expected number of individuals, frequencies of &quot;p&quot; and &quot;q&quot; alleles, p-value,&nbsp;chi-square value and brief&nbsp;information about the significance of results. When the study group is not in a state of genetic equilibrium (p &le; alfa) application will count <strong>F<sub>is</sub> </strong>value to estimate the reason of disturbance. <strong>If observed number of individuals in one of the cells is smaller then 5, application will count additionally chi-square and p-value with Yate`s correction for continuity in this case status of results will be determined on the basis of these values.&nbsp;</strong></p>\r\n\r\n<p style=\"text-align:justify\">More information in&nbsp;<a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; Methods</a></p>\r\n',
        '<p style=\"text-align:center\"><strong>In the case of float numbers please use dots instead of commas!</strong></p>\r\n',
        'Hardy-Weinberg Equilibrium', 1),
       (3, 'Chi-Square tests', '/chi-square-page',
        'Independence chi-square and goodness of fit chi-square tests for contingency tables of any size.',
        'chi-square, Yates\'s correction for continuity, p-value, coefficient of contingency, degrees of freedom, online calculator',
        '<p style=\"text-align:justify\">This tool allows performing&nbsp;tests from chi-square family such as: &quot;independence&quot;&nbsp;and &quot;goodness of fit&quot; chi-square. Users in &quot;independence&quot; chi-square test determine the number of columns and rows (groups and variables). Output values are degrees of freedom (dof), chi-square value, p-value, Yate`s chi-square value, Yate`s p-value and&nbsp;coefficients of contingency values and type.</p>\r\n\r\n<p style=\"text-align:justify\">In &quot;goodness of fit&quot; chi-square users determine the number of groups.&nbsp;In the contingency table are two types of boxes to fill: &quot;expected&quot; and &quot;observed&quot;. Summary values for both rows must be equal.&nbsp;Output values are chi-square value, p-value, Yate`s chi-square, Yate`s p-value,&nbsp;degrees of freedom (dof).</p>\r\n\r\n<p style=\"text-align:justify\">More information in&nbsp;<a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; methods</a></p>\r\n',
        '<p style=\"text-align:center\"><strong>In the case of float numbers please use dots instead of commas!</strong></p>\r\n',
        'Chi-Square Tests', 1),
       (4, 'Polymorphic information content & Heterozygosity', '/pic',
        'Tool to count PIC and H for dominant or codominant markersin both biallelic or multiallelic locus',
        'polymorphic information content (PIC) and heterozygosity (H), genetics markers, dominant marker, codominant marker, biallelic locus, multiallelic locus, online calculator',
        '<p style=\"text-align:justify\">This tool allows determining Polymorphic information content and Heterozygosity values. Users can select two&nbsp;types of marker (dominant or codominant). For codominant marker at first, users determine the number of alleles in the examined locus and then number or frequencies of alleles in the study group (sum of alleles frequencies must equal 1). For a dominant marker, PIC is counted only for a biallelic locus. Users need to fill number or&nbsp;frequency&nbsp;of amplified bands and number or frequency&nbsp;of absence bands in the study group.</p>\r\n\r\n<p style=\"text-align:justify\"><strong>In case of dominant marker, it is possible to count an approximate number of heterozygous individuals based on Hardy-Weinberg law (more information in </strong><a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; Methods</a><strong>)!</strong></p>\r\n',
        '<p style=\"text-align:center\"><strong>In the case of float numbers please use dots instead of commas!</strong></p>\r\n',
        'Polymorphic Information Content (PIC) and Heterozygosity (H).', 1),
       (5, 'Genetic Distance', '/genetic-distance',
        'Tool to estimate genetic distance between n-number of pairs-populations on the basis of alleles frequencies.',
        'genetic distance, dendrograms, distance matrix, online calculator',
        '<p style=\"text-align:justify\">The tool allows determining genetic distance between pairs-populations based on alleles frequencies. At first, users must determine the number&nbsp;of populations, number of loci, type of distance and type of dendrogram to render and then define a number of alleles in each locus. The second step is to fill alleles frequencies&nbsp;in all boxes (sum of&nbsp;alleles frequencies in each locus must equal 1 or 0).</p>\r\n\r\n<p style=\"text-align:justify\">More information in&nbsp;<a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; methods</a></p>\r\n',
        '<p style=\"text-align:center\"><strong>In the case of float numbers please use dots instead of commas!</strong></p>\r\n',
        'Genetic Distance', 1),
       (6, 'About', '/about', 'A little bit information about our team',
        'about, about us, Szymon Miks, Jan Bińkowski, gene calc team',
        '<p><strong>A few words about us:</strong><br />\r\n<br />\r\nThe <strong>&quot;Gene-Calc&quot;</strong> team is created by <strong>students </strong>whose passion is to create programming solutions for research purposes.&nbsp;<br />\r\nOur team is open to cooperate with anyone who is keen on bioscience or has a need for <strong>programming solutions</strong> for fields of <strong>biotechnology </strong>or <strong>data science</strong>.</p>\r\n\r\n<p>&nbsp;</p>\r\n\r\n<p>The scientific supervisor of the project is prof. Marek Kmieć</p>\r\n',
        '', 'About Us', 1),
       (7, 'Contact Us', '/contact', 'Contact us via e-mail', 'Contact Us, affiliation',
        '<p><br />\r\n<strong>Szymon Miks - Developer&nbsp;<br />\r\n84-30, Lębork</strong><br />\r\nUniversity of Technology in Gdańsk, Poland<br />\r\n<a href=\"https://szymonmiks.pl/\" target=\"_blank\">szymonmiks.pl</a></p>\r\n\r\n<hr />\r\n<p><strong>Jan Bińkowski - Researcher<br />\r\n70-721, Szczecin</strong><br />\r\nWest Pomerania University of Technology in Szczecin, Poland</p>\r\n\r\n<hr />\r\n<p>Contact us via&nbsp;e-mail <a href=\"mailto:contact@gene-calc.pl?subject=Gene%20Calc%20-%20contact%20form\">contact@gene-calc.pl</a>&nbsp;or contact form</p>\r\n\r\n<p><strong>How to cite this page:<br />\r\nBińkowski J., Miks S., (2018, September). Gene-Calc [Computer software]. Available from: www.gene-calc.pl.</strong></p>\r\n',
        '', 'Contact gene-calc', 1),
       (8, 'Our donors and cooperators', '/donors', 'Details about our donors and / or cooperators.',
        'donors, helpers, cooperators',
        '<p><strong>The main donors of this project is: </strong></p>\r\n\r\n<ul>\r\n	<li><strong><img alt=\"\" src=\"/static/uploads/aa-logo.png\" style=\"height:20px; width:40px\" /><a href=\"https://www.aabiot.com/\">A&amp;A BIOTECHNOLOGY</a>&nbsp;- is a Polish biotechnology company that develops and produces innovative products for the life science industry.</strong><br />\r\n	&nbsp;</li>\r\n	<li><img alt=\"MiroArt logo\" src=\"data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz48c3ZnIHZlcnNpb249IjEuMSIgaWQ9IldhcnN0d2FfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiIHZpZXdCb3g9IjAgMCAxMDUgOTYuNiIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMTA1IDk2LjY7IiB4bWw6c3BhY2U9InByZXNlcnZlIj48c3R5bGUgdHlwZT0idGV4dC9jc3MiPi5zdDB7ZmlsbDojRTMwNjEzO30uc3Qxe2ZpbGw6I0ZGRkZGRjt9PC9zdHlsZT48Zz48cGF0aCBjbGFzcz0ic3QwIiBkPSJNODYuOSw3Ni40TDg2LjksNzYuNCBNODguNyw3Ni4xYy0wLjgsMC4zLTEuOCwwLjQtMi43LDAuNGwtMjkuNywwYy0xLjMsMC0yLjMtMC4xLTMuMS0wLjVjLTAuNy0wLjQtMS4zLTEtMS42LTEuOWwtMy04LjJjMi4xLDEuMSw0LjgsMS43LDcuOCwxLjNsMCwwYzAuNC0wLjEsMC43LTAuMSwxLTAuMmMwLjUsMC45LDEuMywxLjYsMi4yLDIuM2MxLjIsMC44LDIuOCwxLjQsNC40LDEuNGgxNC4zaDIuMWwtMC43LTEuOEw2Ny45LDM3LjhMNjYuNCwzNGwtMS40LDMuOGwtNi42LDE4LjRjLTAuMywwLjgtMC42LDEuNS0xLDIuMWMtMC40LDAuNS0wLjcsMS0xLjIsMS4zYy0wLjgsMC42LTEuOCwxLTIuOCwwLjljLTAuOCwwLTEuNy0wLjMtMi4zLTFjLTAuMy0wLjMtMC42LTAuNy0wLjctMS4yYy0wLjItMC41LTAuMy0xLTAuMy0xLjZjMC0wLjEsMC0wLjIsMC0wLjRjMC0wLjEsMC0wLjIsMC0wLjNjMCwwLDAtMC4xLDAtMC4yYzAtMC4xLDAsMCwwLTAuMWMwLTAuMSwwLjEtMC4zLDAuMi0wLjdjMC4xLTAuMiwwLjItMC41LDAuMy0wLjlsMCwwbDExLjItMzEuM2MwLjMtMC43LDAuNS0xLjMsMC45LTEuOGMwLjMtMC41LDAuNi0wLjgsMS4xLTEuMWMwLjQtMC4zLDAuOS0wLjYsMS4zLTAuN2MwLjQtMC4xLDAuOC0wLjIsMS4zLTAuMmMxLjEsMCwxLjksMC4yLDIuNiwwLjdjMC43LDAuNCwxLjIsMS4xLDEuNiwyLjF2MGwxOSw1MGwwLDBjMC4yLDAuNywwLjQsMS4zLDAuNSwyYzAuMSwwLjYsMCwxLjMtMC40LDEuOHYwbDAsMGMtMC4xLDAuMS0wLjIsMC4yLTAuMywwLjNDODkuMSw3NS45LDg4LjksNzYsODguNyw3Ni4xIi8+PHBhdGggY2xhc3M9InN0MSIgZD0iTTg2LjksNzYuNEw4Ni45LDc2LjQgTTg4LjcsNzYuMWMtMC44LDAuMy0xLjgsMC40LTIuNywwLjRsLTI5LjcsMGMtMS4zLDAtMi4zLTAuMS0zLjEtMC41Yy0wLjctMC40LTEuMy0xLTEuNi0xLjlsLTMtOC4yYzIuNCwxLjIsNS4xLDEuNiw3LjgsMS4yYzAuNC0wLjEsMC44LTAuMSwxLTAuMmMwLjUsMC45LDEuMywxLjYsMi4yLDIuM2MxLjIsMC44LDIuOCwxLjQsNC40LDEuNGgxNC4zaDIuMWwtMC43LTEuOEw2Ny45LDM3LjhMNjYuNCwzNGwtMS40LDMuOGwtNi42LDE4LjRjLTAuMywwLjgtMC42LDEuNS0xLDIuMWMtMC40LDAuNS0wLjgsMS0xLjIsMS4zYy0wLjgsMC42LTEuOCwxLTIuOCwwLjljLTAuOCwwLTEuNy0wLjMtMi4zLTFjLTAuMy0wLjMtMC42LTAuNy0wLjctMS4yYy0wLjItMC41LTAuMy0xLTAuMy0xLjZjMC0wLjIsMC0wLjUsMC0wLjdjMC0wLjMsMC4xLTAuNiwwLjItMC45YzAuMS0wLjIsMC4yLTAuNSwwLjMtMC45bDExLjItMzEuM2MwLjMtMC43LDAuNS0xLjMsMC45LTEuOGMwLjMtMC41LDAuNi0wLjgsMS4xLTEuMWMwLjQtMC4zLDAuOS0wLjYsMS4zLTAuN2MwLjQtMC4xLDAuOC0wLjIsMS4zLTAuMmMxLjEsMCwxLjksMC4yLDIuNiwwLjdjMC43LDAuNCwxLjIsMS4xLDEuNiwyLjFsMTksNTBoMGMwLjIsMC43LDAuNCwxLjMsMC41LDJjMC4xLDAuNiwwLDEuMy0wLjQsMS44djBjLTAuMSwwLjEtMC4yLDAuMi0wLjMsMC4zQzg5LjEsNzUuOSw4OC45LDc2LDg4LjcsNzYuMSBNMzYuNiwzOC4xbDEzLjQsMzYuNGMxLDIuNywzLjIsMy40LDYuMiwzLjRsMjkuNywwYzQuMywwLDYuNy0xLjgsNC45LTYuNmwtMTkuMS01MGMtMS0yLjQtMi44LTMuNi01LjYtMy42Yy0xLjIsMC0yLjQsMC40LTMuNSwxLjJjLTEuMSwwLjgtMS44LDItMi40LDMuNkw0OS4yLDUzLjhjLTAuMywwLjktMC41LDEuNS0wLjUsMS43YzAsMC4yLTAuMSwwLjYtMC4xLDEuMmMwLDEuNSwwLjUsMi44LDEuNCwzLjdjMC45LDAuOSwyLjEsMS40LDMuNSwxLjRjMS4zLDAsMi41LTAuNCwzLjctMS4zYzEuMS0wLjgsMi0yLjIsMi43LTRsNi42LTE4LjRjMy45LDEwLjMsNy45LDIwLjYsMTEuOCwzMC44SDY0Yy0yLjksMC01LjItMi01LjktNGMtMC4zLDAuMS0wLjgsMC4zLTIuMSwwLjVjLTcuMSwwLjktMTEuOC00LjEtMTEuOC04LjFjMC0xLjIsMC42LTMuNSwxLjEtNS4ybDQuMi0xMS44TDQyLDIxLjFjLTEtMi43LTMtMy45LTUuNC0zLjljLTEuMywwLTIuNSwwLjUtMy43LDEuNGMtMS4xLDAuOS0yLjEsMi40LTIuOCw0LjNMMTMuNiw3MS41Yy0wLjIsMC41LTAuNCwxLjEtMC41LDEuN2MtMC4xLDAuNi0wLjEsMS0wLjEsMS40YzAsMS41LDAuNSwyLjcsMS40LDMuNmMwLjksMC45LDIsMS40LDMuMywxLjRjMywwLDUuMS0xLjYsNi4zLTQuOUwzNi42LDM4LjF6Ii8+PHBhdGggZD0iTTM2LjYsMzguMWwxMy40LDM2LjRjMSwyLjcsMy4yLDMuNCw2LjIsMy40bDI5LjcsMGM0LjMsMCw2LjctMS44LDQuOS02LjZsLTE5LjEtNTBjLTAuOS0yLjQtMi44LTMuNi01LjYtMy42Yy0xLjIsMC0yLjQsMC40LTMuNSwxLjJjLTEuMSwwLjgtMS44LDItMi40LDMuNkw0OS4yLDUzLjhjLTAuMywwLjktMC41LDEuNS0wLjUsMS43YzAsMC4yLTAuMSwwLjYtMC4xLDEuMmMwLDEuNiwwLjUsMi44LDEuNCwzLjdjMC45LDAuOSwyLjEsMS40LDMuNSwxLjRjMS4zLDAsMi41LTAuNCwzLjctMS4zYzEuMS0wLjgsMi0yLjIsMi43LTRsNi42LTE4LjRjMy45LDEwLjMsNy45LDIwLjYsMTEuOCwzMC44SDY0Yy0yLjksMC01LjItMi01LjktNGMtMC4zLDAuMS0wLjgsMC4zLTIuMSwwLjVjLTcuMSwwLjktMTEuOC00LjEtMTEuOC04LjJjMC0xLjIsMC42LTMuNSwxLjEtNS4ybDQuMi0xMS44TDQyLDIxLjFjLTEtMi43LTIuOS0zLjktNS40LTMuOWMtMS4zLDAtMi41LDAuNS0zLjcsMS40Yy0xLjEsMC45LTIuMSwyLjQtMi44LDQuM0wxMy42LDcxLjVjLTAuMiwwLjUtMC40LDEuMS0wLjUsMS43Yy0wLjEsMC42LTAuMSwxLTAuMSwxLjRjMCwxLjUsMC41LDIuNywxLjQsMy42YzAuOSwwLjksMiwxLjQsMy4zLDEuNGMzLDAsNS4xLTEuNiw2LjMtNC45TDM2LjYsMzguMXogTTY2LjcsNTIuNWMtMS4xLDIuOS0yLjYsNy40LTQuNSwxMGMwLjMsMS4zLDEuNSwyLjMsMy4zLDIuM2g2LjNMNjYuNyw1Mi41eiBNMjUuOSwyMC40YzAuOC0yLjMsMi4yLTQuMSw0LjMtNS41YzItMS4zLDQuMi0yLDYuNS0yYzIuMSwwLDQsMC41LDUuOCwxLjZjMS43LDEsMywyLjYsMy44LDQuN2w1LjYsMTQuMWw0LjUtMTMuNWMwLjctMi4xLDEuOS0zLjcsMy43LTQuN2MxLjgtMS4xLDMuOC0xLjYsNi0xLjZjMi4zLDAsNC4zLDAuNSw2LjIsMS42YzEuOSwxLDMuMiwyLjYsNCw0LjZMOTYsNzAuOWMyLjgsNy40LTIuMywxMS4yLTkuMSwxMS4ySDU2LjNjLTUuNiwwLTguNi0wLjgtMTAuNy02LjRsLTguOC0yNGwtOC43LDI1Yy0wLjcsMi4zLTIsNC0zLjgsNS4yYy0xLjgsMS4yLTMuOSwxLjgtNi4xLDEuOGMtMi43LDAtNS4xLTAuOC03LTIuNGMtMi0xLjYtMy0zLjgtMy02LjZjMC0xLjMsMC4yLTIuNiwwLjctNEwyNS45LDIwLjR6Ii8+PC9nPjwvc3ZnPg==\" style=\"height:32px; width:35px\" /><a href=\"https://www.miroart.pl/\" target=\"_blank\">MiroArt Strony Internetowe</a>&nbsp;-&nbsp;<strong>specialized&nbsp;in&nbsp;creating and managing websites for 15 years.</strong></li>\r\n</ul>\r\n\r\n<p><strong>Cooperators:</strong></p>\r\n\r\n<ul>\r\n	<li><a href=\"http://www.e-biotechnologia.pl/Artykuly/Gene-Calc/\" target=\"_blank\"><strong><img alt=\"\" src=\"/static/uploads/biotech.png\" style=\"height:35px; width:40px\" />&nbsp;</strong>e-biotechnologia.pl</a>&nbsp;-&nbsp;<strong>information service associated with biotechnology.</strong></li>\r\n</ul>\r\n\r\n<p>​​​​</p>\r\n',
        '', 'Our donors and cooperators', 1),
       (9, 'Dot plot', '/sequences-analysis-tools/dot-plot',
        'This tools allows to generate alignment, with dot plot for for compared sequences.',
        'dot plot online, sequences alignment',
        '<p>Tool to the graphic presentation of sequences alignment. Users&nbsp;must choose the type of input data: raw sequences or <strong>GenBank ID</strong>&nbsp;used to the calculation. Output values are: sequences length [bp], coverage [%], average identity [%], fragmental identity [%], alignment, score and dot plot.&nbsp;</p>\r\n\r\n<p>More information in&nbsp;<a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; Methods</a></p>\r\n',
        '<p>Because of upcoming changes to the NCBI databases, tools connected with Entrez may be unavailable.</p>\r\n',
        'Dot plot', 1),
       (10, 'Consensus Sequence', '/sequences-analysis-tools/consensus-sequence',
        'This tool allows to obtain consensus sequence based on large data of sequences.', 'Consensus Sequence online',
        '<p>Tool to determine the consensus sequence based on numerous sequences. Users must choose type of input data: raw&nbsp;sequences in <strong>FASTA </strong>format, sequences from <strong>FASTA</strong> file or sequences downloaded from <strong>GenBank</strong>&nbsp;by means of their ID&nbsp;numbers. Output values are: approximate sequence length&nbsp;and a consensus sequence.&nbsp;The threshold value&nbsp;is&nbsp;the frequency (in range 0-1) of nucleotide on n-position which must be obtained to consider the nucleotide as a conservative.</p>\r\n\r\n<p>More information in&nbsp;<a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; methods</a></p>\r\n',
        '<p>Because of upcoming changes to the NCBI databases, tools connected with Entrez may be temporary unavailable.</p>\r\n',
        'Tool to obtain consensus sequence.', 1),
       (11, 'Sequences Tools', '/sequences-analysis-tools/sequences-tools',
        'Tools to perform standard sequence transformations such as: transcription, translation, reverse ...',
        'sequences tools, DNA tools, reverse sequence, transcription, translation, complement, reverse complement.',
        '<p>This tool allows transforming&nbsp;one or numerous sequences in <strong>FASTA </strong>format.&nbsp;</p>\r\n\r\n<p>More information in&nbsp;<strong><a href=\"https://gene-calc.pl/materials-and-methods\">Materials &amp; Methods</a></strong></p>\r\n\r\n<p>&nbsp;</p>\r\n',
        '', 'Tools to deal with sequences.', 1);
INSERT INTO `pages` (`id`, `name`, `slug`, `seo_desc`, `seo_keywords`, `text`, `desc`, `seo_title`, `is_active`)
VALUES (12, 'Materials & Methods', '/materials-and-methods',
        'This page is intended to provide a brief, theoretical introduction to all calculations used by application.',
        'Yates\'s correction for continuity, PIC and H, chi-square, p-value, research hypothesis, coefficient of contingency',
        '<p style=\"margin-right:0cm; text-align:justify\"><span style=\"font-size:18px\"><strong>Table of</strong></span><span style=\"font-size:18px\"><strong>&nbsp;contents:</strong></span></p>\r\n\r\n<ul>\r\n	<li style=\"text-align:justify\"><a href=\"#Hardy-Weinberg\"><span style=\"font-size:16px\">Hardy-Weinberg Equilibrium</span></a></li>\r\n	<li style=\"text-align:justify\"><a href=\"#Chi-square\"><span style=\"font-size:16px\">Chi-square</span></a></li>\r\n	<li style=\"text-align:justify\"><a href=\"#PIC &amp; H\"><span style=\"font-size:16px\">PIC &amp; H</span></a></li>\r\n	<li style=\"text-align:justify\"><a href=\"#Genetic distance\"><span style=\"font-size:16px\">Genetic distance</span></a></li>\r\n	<li style=\"text-align:justify\"><a href=\"#seq\"><span style=\"font-size:16px\">Sequences analysis tools</span></a></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><a href=\"#Ref\">References</a>&nbsp; </span></li>\r\n</ul>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><a id=\"Hardy-Weinberg\" name=\"Hardy-Weinberg\"></a></span><span style=\"font-size:18px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\"> Hardy&ndash;Weinberg equilibrium</span></span></strong></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Input values (number of homozygous, heterozygous and rare homozygous individuals) in integers form, are used to count alleles frequencies - &quot;p&quot; and &quot;q&quot; values, by means of Hardy-Weinberg principle the expected numbers of homozygous, heterozygous and rare homozygous individuals are obtained. Observed and expected values are compared by the chi-square goodness of fit test.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">If observed number of individuals in one of the cells is smaller then 5, application will count additionally chi-square and p-value with Yate`s correction for continuity in this case status of results will be determined on the basis of these values.&nbsp;</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">If at the used level of significance, there are statistically important differences, application will count the fixation index - Fis, positive values may means a reduction of heterozygosity, caused by for example Wahlund effect, inbreed or genetic drift. While negative Fis value means the excess of heterozygotes, caused by for example breeding selection or bottleneck effect.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\"><strong>F<sub>IS</sub> = 1 - (observed heterozygosity/expected heterozygosity)</strong></span></span></span></p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><span style=\"font-size:18px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">&nbsp;<a id=\"Chi-square\" name=\"Chi-square\">Chi-square tests</a></span></span></strong></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Chi-square test family is a statistical instrument which allows examining the relation between variables. The chi-square tests must be used with assumptions:</span></span></span></p>\r\n\r\n<ol>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">test group must be sufficiently large.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">data should be maximally randomly drawn from a population.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">values in cells are adequate only when: (1) no more than 20% of the expected values are smaller than five; (2) there are no cells with zero.</span></span></span></li>\r\n</ol>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">Independence (Associations) chi-square</span></span></strong></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Research hypothesis:&nbsp;<br />\r\nH</span><sub><span style=\"color:#212529\">0</span></sub><span style=\"color:#212529\">: There is no association between variables.<br />\r\nH</span><sub><span style=\"color:#212529\">a</span></sub><span style=\"color:#212529\">&nbsp;: There is&nbsp;an association between variables.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">The expected values for independence chi-square and chi-square value are counted following formulas:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><img alt=\"\" src=\"/static/uploads/expected_values.png\" style=\"height:62px; margin-left:350px; margin-right:350px; width:108px\" />&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;<img alt=\"\" src=\"/static/uploads/chi_square.png\" style=\"height:50px; margin-left:350px; margin-right:350px; width:156px\" /></span></p>\r\n\r\n<ul>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">E<sub>ij</sub>&nbsp;= formula to count expected values for the ith row and jth column.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">T<sub>i</sub>&nbsp;= sum of values in the ith row</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">T<sub>j</sub>&nbsp;= sum of values in the jth column</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">N = sum of values in all rows or columns</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">O<sub>i</sub>&nbsp;&ndash; observed value&nbsp;</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">E<sub>i</sub>&nbsp;&ndash; expected value</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">n &ndash; a set of all cells</span></span></span></li>\r\n</ul>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">If the values in the cell do not fulfil third condition (look at <a href=\"#Chi-square\">chi-square assumptions</a>), we recommend to use chi-square with Yates&#39;s correction for continuity, following formula:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\"><img alt=\"\" src=\"/static/uploads/chi_square_yates.png\" style=\"float:left; height:69px; margin-left:350px; margin-right:350px; width:254px\" /></span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">It is important to note that this correction is adequate for 2x2 contingency tables, otherwise more accurate results may be obtained by combining columns or rows. Degrees of freedom are counted by dof = (r &minus; 1) * (c &minus; 1)&nbsp;<br />\r\nr &ndash; number of rows (variables)&nbsp;<br />\r\nc &ndash; number of columns (groups)</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\"><strong>The coefficient of contingency</strong> are measures of association, they can be used to estimate the strength of a relationship between variables.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">A coefficient of contingency are counted following:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">- for 2x2 (1 dof) tables by means of Phi-coefficient:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\"><img alt=\"\" src=\"/static/uploads/Phi.jpg\" style=\"height:110px; margin-left:350px; margin-right:350px; width:174px\" /></span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">- for tables with dof &gt; 1 using Crammer`s V coefficient</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><img alt=\"\" src=\"/static/uploads/crammer.jpg\" style=\"height:103px; margin-left:350px; margin-right:350px; width:195px\" /></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">N, n - sum of values in row or columns</span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">m - smaller of the number of columns or rows</span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Range of Coefficient of contingency value is between 0.0 - 1.0.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">Chi-square goodness of fit test:</span></span></strong></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Allows to examine the compliance of the observed and expected values.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Research hypothesis:&nbsp;<br />\r\nH</span><sub><span style=\"color:#212529\">0</span></sub><span style=\"color:#212529\">: The observed values are equal to&nbsp;theoretical values (expected).&nbsp;<br />\r\nH</span><sub><span style=\"color:#212529\">a</span></sub><span style=\"color:#212529\">&nbsp;: The observed values are not equal to theoretical values (expected).</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Degrees of freedom are counted by:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">dof = k &minus; 1</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">k - number of columns (groups)</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Total values must be the same for both observed and expected frequencies, expected frequencies less than 1 are not allowed.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Part of calculations in this module (as well as in Hardy-Weinberg module) are executed by modified&nbsp;&quot;</span></span>scipy.stats.chi2_contingency&quot;<span style=\"background-color:white\"><span style=\"color:#212529\">&nbsp;<a href=\"https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html\">Python library</a>.</span></span></span></p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><a name=\"PIC &amp; H\"><span style=\"font-size:18px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">&nbsp; </span></span></strong></span></a></p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><a name=\"PIC &amp; H\"><span style=\"font-size:18px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">PIC and H</span></span></strong></span></a></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Heterozygosity (H) is a parameter indicating the average frequency of&nbsp;heterozygous individuals occurrence. Polymorphic information content (PIC), as well as Heterozygosity, is a measure of locus polymorphism counted for markers used in linkage analysis.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">PIC &amp; H are counted for bi,- or multi,- allelic locus following formulas:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">&nbsp;&nbsp;<img alt=\"\" src=\"/static/uploads/pic.png\" style=\"height:67px; margin-left:350px; margin-right:350px; width:348px\" /></span></span></span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\"><img alt=\"\" src=\"/static/uploads/h.png\" style=\"height:63px; margin-left:400px; margin-right:400px; width:147px\" /></span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">P</span><sub><span style=\"color:#212529\">i</span></sub><span style=\"color:#212529\">&nbsp;, P</span><sub><span style=\"color:#212529\">j</span></sub><span style=\"color:#212529\">&nbsp;are the frequency of the i<sub>th</sub> and j<sub>th</sub> allele (float format),&nbsp;<br />\r\nl &ndash; number of alleles</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">PIC for dominant marker (inputs are in form of dichotomous variables 1/0), can be calculated using formula:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">PIC = 1&minus; [fi</span><sup><span style=\"color:#212529\">2</span></sup><span style=\"color:#212529\">&nbsp;+ (1&minus; fi)</span><sup><span style=\"color:#212529\">2</span></sup><span style=\"color:#212529\">]</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Where fi is frequency of amplified band (1) and (1 - fi) is frequency of absence of band (0) in study group.</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">for codominant marker range of PIC value is between 0.0-1.0</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">for dominant marker range of PIC value is between 0.0-0.5</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<blockquote>\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"color:#c0392b\"><strong><span style=\"font-size:16px\"><span style=\"background-color:white\">Important note!</span></span></strong></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"color:#c0392b\"><span style=\"font-size:16px\">In case of dominant marker it is possible to count an approximate number of heterozygous individuals based on Hardy-Weinberg law:</span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"color:#c0392b\"><span style=\"font-size:16px\">If <em>q</em><sup>2</sup> is a frequency of homozygous genotype (absence band frequency) then (<em>q</em><sup>2</sup>)<sup>1/2</sup> is <em>q</em>&nbsp;allele frequency. In case of biallelic locus <em>p&nbsp;</em>+ <em>q&nbsp;</em>= 1, so <em>p</em>&nbsp;= 1 &ndash; <em>q</em>. Based on counted alleles frequencies we could estimate the expected number of genotypes from Hardy-Weinberg law.</span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"color:#c0392b\"><span style=\"font-size:16px\">Above-mentioned calculations allow counting more accurate&nbsp; PIC value based on results obtained with dominant markers (with the assumption that study group is in a state of genetic balance).</span></span></p>\r\n</blockquote>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><a name=\"Genetic distance\"><span style=\"font-size:18px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">&nbsp;</span></span></strong></span></a></p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><a name=\"Genetic distance\"><span style=\"font-size:18px\"><strong><span style=\"background-color:white\"><span style=\"color:#212529\">Genetic distance</span></span></strong></span></a></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Genetic distance is a measure of differences between pairs of populations, there are a couple of methods allows to estimate differences, based on alleles frequency in examined loci, such as:</span></span></span></p>\r\n\r\n<ol>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Standard genetic distance (Nei, 1972).</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Geometric genetic distance (Nei, 1978).</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Genetic distance based on STR loci polymorphism (Nei and Takazaki, 1983).</span></span></span></li>\r\n</ol>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">1) Standard genetic distance is used to determine the normalized differences between populations when the individual codon changes are independent and they are&nbsp;compliance with Poisson distribution,&nbsp;by means of formula:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><img alt=\"https://gene-calc.pl/static/img/formulas/genetic-distance.png\" src=\"/static/uploads/111111111111111.png\" style=\"height:200px; margin-left:350px; margin-right:350px; width:300px\" /></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">2) Geometric genetic distance allows to determine differences between populations when loci differ in rate of codon changes (in this case standard genetic distance may be underestimated). Geometric distance is calculated following formula:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><img alt=\"https://gene-calc.pl/static/img/formulas/geometric-distance.png\" src=\"/static/uploads/w222222.png\" style=\"height:200px; margin-left:350px; margin-right:350px; width:300px\" /></span></p>\r\n\r\n<blockquote>\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><strong><span style=\"color:#e74c3c\"><span style=\"font-size:18px\"><span style=\"background-color:#ffffff\">Caution !&nbsp;&nbsp; </span></span></span></strong></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"color:#e74c3c\"><span style=\"font-size:18px\"><span style=\"background-color:#ffffff\">When some of the allele frequencies are equal to 0, geometric distance is impossible to calculate, because J</span><sub><span style=\"background-color:#ffffff\">x</span></sub><span style=\"background-color:#ffffff\">, J</span><sub><span style=\"background-color:#ffffff\">y</span></sub><span style=\"background-color:#ffffff\"> and J</span><sub><span style=\"background-color:#ffffff\">xy</span></sub><span style=\"background-color:#ffffff\"> are geometric means of j</span><sub><span style=\"background-color:#ffffff\">x</span></sub><span style=\"background-color:#ffffff\">, j</span><sub><span style=\"background-color:#ffffff\">y</span></sub><span style=\"background-color:#ffffff\"> and j</span><sub><span style=\"background-color:#ffffff\">xy</span></sub><span style=\"background-color:#ffffff\"> values. So in case if one of&nbsp;j</span><sub><span style=\"background-color:#ffffff\">x&nbsp;</span></sub><span style=\"background-color:#ffffff\">or&nbsp;j</span><sub><span style=\"background-color:#ffffff\">y</span></sub><span style=\"background-color:#ffffff\">&nbsp;or&nbsp;j</span><sub><span style=\"background-color:#ffffff\">xy&nbsp;</span></sub><span style=\"background-color:#ffffff\">= 0&nbsp;</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"color:#e74c3c\"><span style=\"font-size:18px\"><span style=\"background-color:#ffffff\">J</span><sub><span style=\"background-color:#ffffff\">x </span></sub><span style=\"background-color:#ffffff\">or&nbsp;J</span><sub><span style=\"background-color:#ffffff\">y</span></sub><span style=\"background-color:#ffffff\">&nbsp;or&nbsp;J</span><sub><span style=\"background-color:#ffffff\">xy&nbsp; </span></sub><span style=\"background-color:#ffffff\">also&nbsp;</span><span style=\"background-color:#ffffff\">will be equal to 0, it makes that application will try to divide by 0 or count -ln(0). In this case app return error: &quot;The quantity or quality of the data is inappropriate!&quot;.</span></span></span></p>\r\n</blockquote>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">3) Genetic distance could be an estimate based on short tandem repeats (STR) polymorphism (Nei and Takazaki, 1983). This model of genetic distance estimation was considered more effective than other methods. It is calculated following formula:</span></span></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><img alt=\"https://gene-calc.pl/static/img/formulas/takazaki-nei.png\" src=\"https://gene-calc.pl/static/img/formulas/takazaki-nei.png\" style=\"height:91px; margin-left:350px; margin-right:350px; width:254px\" /></span></p>\r\n\r\n<ul>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">D - standard genetic distance</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">D` - geometric genetic distance</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">D<sub>A</sub>&nbsp;- genetic distance based on STR polymorphism</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">j<sub>X</sub>- the sum of allele frequency squares in &quot;j&quot; locus in X population.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">j<sub>Y</sub>- the sum of allele frequency squares in &quot;j&quot; locus in Y population.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">j<sub>XY</sub>- the sum of allele frequency product between X and Y population in &quot;j&quot; locus.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">J<sub>X</sub>-the arithmetic or geometric (J`) mean (depending on the type of distance) the j<sub>X</sub>&nbsp;values for the pair-population.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">J<sub>Y</sub>- the arithmetic or geometric (J`) mean (depending on the type of distance) the j<sub>Y</sub>&nbsp;values for the pair-population.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">J<sub>XY</sub>- the arithmetic or geometric (J`) mean (depending on the type of distance) the j<sub>XY</sub>&nbsp;values for the pair-population.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">X<sub>ij</sub>&nbsp;&amp; Y<sub>ij</sub>&nbsp;frequencies of &quot;i&quot; STR in &quot;j&quot; locus in pair-population X,Y</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">r - a number of analyzed STR loci</span></span></span></li>\r\n</ul>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">The results of genetic distance estimation are presented in the form of distance matrix and dendrogram. The application offers the possibility to select&nbsp;several types of methods to construct dendrograms such as unweighted pair group method with arithmetic mean (UPGMA), weighted pair group method with averaging (WMPGA), centroid linkage clustering (UPGMC), weighted pair-group centroid clustering (WPGMC), single-linkage clustering and complete-linkage clustering. The construction of dendrograms is performed by the Python library &quot;scipy.cluster.hierarchy.linkage&quot; lib documentation and theoretical information are available from&nbsp;<a href=\"https://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.hierarchy.linkage.html\">Python library</a></span></span></span></p>\r\n\r\n<p><strong><span style=\"font-size:18px\"><a id=\"seq\" name=\"seq\"></a></span></strong></p>\r\n\r\n<p><strong><span style=\"font-size:18px\">Sequences analysis tools</span></strong></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">This module contains a few&nbsp;tools dedicated for work&nbsp;with nucleotide or amino acid sequences. Based on <strong><a href=\"http://biopython.org/DIST/docs/tutorial/Tutorial.html\" target=\"_blank\">Biopython </a></strong>lib and&nbsp;<strong><a href=\"https://www.drive5.com/muscle/\" target=\"_blank\">Muscle</a>.</strong></span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><strong>1)</strong> <strong>Dot plot </strong>allows </span><span style=\"font-size:16px\">to compare</span><span style=\"font-size:16px\"> two sequences. Input values: names and raw sequences (only A, C, T/U, G and N&nbsp;characters&nbsp;in sequences are acceptable) or <strong>GenBank </strong>ID: e.g.&nbsp;KY783590.3</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">output values </span><span style=\"font-size:16px\">are:</span><span style=\"font-size:16px\"> classic alignment, dot plot, and parameters:&nbsp;</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">- Score: one point for every match,&nbsp;max score value = length of </span><span style=\"font-size:16px\">shorter</span><span style=\"font-size:16px\"> sequence.</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">- Coverage:&nbsp;percentage of sequences coverage.</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">-&nbsp;Average identity: </span><span style=\"font-size:16px\">number</span><span style=\"font-size:16px\"> of matches divided by&nbsp;average sequences length x 100.</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">-&nbsp;Fragmental identity:&nbsp;number of matches in coverage fragment of sequences x 100.</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><strong>Important note</strong>&nbsp;&quot;In case of short sequences (up to 20 bp) dot plot will not generate.&quot;</span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><strong>2)&nbsp;Consensus Sequence&nbsp;</strong>allows </span><span style=\"font-size:16px\">to determine</span><span style=\"font-size:16px\"> consensus sequence based on numerous sequences.</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">Acceptable input format:</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">a) sequences in <strong>FASTA </strong>format e.g.&nbsp;</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">&gt;XYZ</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">gtcgtg</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">&gt;ZYX</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">actga</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">b)<strong> GenBank IDs, </strong>each ID in second line e.g.</span></p>\r\n\r\n<pre style=\"margin-left:40px\">\r\n<span style=\"font-size:16px\">Z78532.1</span></pre>\r\n\r\n<pre style=\"margin-left:40px\">\r\n<span style=\"font-size:16px\">Z78533.1</span></pre>\r\n\r\n<pre style=\"margin-left:40px\">\r\n<span style=\"font-size:16px\">Z78529.1</span></pre>\r\n\r\n<p style=\"margin-left:40px\"><span style=\"font-size:16px\">...</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">c) <strong>FASTA</strong> file (<a href=\"https://raw.githubusercontent.com/biopython/biopython/master/Doc/examples/ls_orchid.fasta\" target=\"_blank\">example</a>)</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">In case of Fasta file upload and sequences downloaded from GenBank, app return file containing </span><span style=\"font-size:16px\">consensus</span><span style=\"font-size:16px\"> sequence in Fasta format.</span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">The result&nbsp;is a consensus sequence. If more than threshold value (frequency) of elements in all input sequences in <strong>n </strong>position are the same, this element in position <strong>n </strong>is consider as consensus element in sequence, if no nucleotide appears more often than&nbsp;threshold value in used&nbsp;sequences in a consensus sequence will be &quot;N&quot; character, if sequences have different length and if less than threshold value&nbsp;have any nucleotide in n<strong>&nbsp;</strong>position in consensus sequence will be &quot;-&quot;&nbsp;character.</span></p>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\"><strong>3)&nbsp;Sequences Tools</strong></span></p>\r\n\r\n<p style=\"margin-left:40px; margin-right:0cm; text-align:justify\"><span style=\"font-size:16px\">Allows </span><span style=\"font-size:16px\">to transform</span><span style=\"font-size:16px\"> (reverse, complement, reverse-complement, transcription, translation) one or numerous sequences in <strong>FASTA </strong>format.</span></p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><strong><span style=\"font-size:16px\">We recommend <a href=\"https://www.megasoftware.net/\" target=\"_blank\">MEGA&nbsp;</a>for handling FASTA files.</span></strong><br />\r\n&nbsp;</p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><a name=\"Ref\"><strong><span style=\"font-size:18px\"><span style=\"background-color:white\"><span style=\"color:#212529\">&nbsp;</span></span></span></strong></a></p>\r\n\r\n<p style=\"margin-right:0cm; text-align:justify\"><a name=\"Ref\"><strong><span style=\"font-size:18px\"><span style=\"background-color:white\"><span style=\"color:#212529\"> </span></span></span><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">References:</span></span></span></strong></a></p>\r\n\r\n<ul>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Sa&acute;ndor Nagy, Pe&acute;ter Poczai, Istva&acute;n Cerna&acute;k, Ahmad Mousapour Gorji, Ge&acute;za Hegedus , Ja&acute;nos Taller, (2012): PICcalc: An Online Program to Calculate Polymorphic Information Content for Molecular Genetic Studies, Biochemical Genetics 50: 670&ndash;672.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Yu.V. Chesnokov, A.M. Artemyeva (2015): Bioinformatics and math statistics, Agricultural Biology, 50 (5): 571-578.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Pilar Soengas, Pablo Velasco, Guillermo Padilla, Amando Ordas, Maria Elena Cartea (2006). Genetic relationships among Brassica napus crops based on SSR markers, HORT SCIENCE 41, 1195&ndash;1199.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Sorana D. Bolboacă , Lorentz J&auml;ntschi, Adriana F. Sestraş , Radu E. Sestraş and Doru C. Pamfil, (2011): Pearson-Fisher Chi-Square Statistic Revisited, Information, 2, 528-545.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Yates, F. (1934) Contingency tables involving small numbers and the &chi;2 test. Journal of the Royal Statistical Society, Suppl.1, 217&ndash;235.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Zeynel Dalkılı&ccedil; , H. Osman Mestav, Gonca G&uuml;nver-Dalkılı&ccedil; and Hilmi Kocataş (2011): Genetic diversity of male fig (Ficus carica caprificus L.) genotypes with random amplified polymorphic DNA (RAPD) markers, African Journal of Biotechnology, 10 (4): 519-526.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Antony Ugoni, Bruce F. Walker (1995): The Chi square test: an introduction, Chiropractors and Osteopaths Musculo-Skeletal Interest Group. 4. 61-4.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Quinnipiac University. Maaning of Pearson&rsquo;s. Retrieved Jun 20, 2016 from: http://faculty.quinnipiac.edu/libarts/polsci/Statistics.html Yule, G. U. (1912). J. R. Statist. Soc., 75, 76&ndash;642.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Chi Square Based Measures, from&nbsp;<a href=\"http://uregina.ca/~gingrich/ch11a.pdf\" target=\"_blank\"><span style=\"color:#007bff\">http://uregina.ca/~gingrich/ch11a.pdf</span></a></span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Robin S.&nbsp;Waples (2015): Testing for Hardy&ndash;Weinberg Proportions: Have We Lost the&nbsp;Plot ?, Journal of Heredity, 106 (1): 1&ndash;19.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">M. Nei, R. K. Chesser (1983): Estimation of fixation indices and gene diversites, Annal of Human Genetics, 47, 253-259.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">R. K. Chesse (1991): Influence of gene flow and breeding tactics on gene diversity within populations. Genetics, 129&nbsp;(2):&nbsp;573-583.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Bolesław Żuk, Heliodor Wierzbicki, Magdalena Zatoń-Dobrowolska, Zofia Kulisiewicz. Genetyka Populacji i Metody Hodowlane, Powszechne Wydawnictwo Rolnicze i Leśne, Warszawa, 2011.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Rold&aacute;n-Ruiz, I., van Euwijk, F., Gilliland, T. et al.&nbsp;(2001): A comparative study of molecular and morphological methods of describing relationships between perennial ryegrass (Lolium perenne L.) varieties, Theoretical and Applied Genetics, &nbsp;103(8): 1138&ndash;1150.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Nei M.,Tajima F. and Tateno Y. (1983): Accuracy of estimated phylogenetic trees from molecular data. II. Gene frequency data. Journal of Molecular Evolution, 19, 153-170.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Carl E. Hildebrand, David C. Torney, and Robert P. Wagner (1992): Informativeness of Polymorphic DNA Markers, Los Alamos Science, 20, 100-102.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Li Jin and Ranajit Chakraborty (1994): Estimation of Genetic Distance and Coefficient of Gene Diversity from SingleProbe Multilocus DNA Fingerprinting Data, Molecular Biology and Evolution, 11(1): 120-127.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Masatoshi Nei (1978): The Theory of Genetic Distance and Evolution of Human Races, Japan Journal of Human Genetics, 23, 341-369</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Masatoshi Nei and Naoko Takazaki (1996): The Root of the Phylogenetic Tree of Human Populations, 13(1): 170-177.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">Masatoshi Nei (1972): Genetic distance between populations, The American Naturalist 106, 283-292.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\"><span style=\"background-color:white\"><span style=\"color:#212529\">J. De Riek &middot; E. Calsyn &middot; I. Everaert &middot; E. Van Bockstaele M. De Loose (2001), AFLP based alternatives for the assessment of Distinctness, Uniformity and Stability of sugar beet varieties, Theor Appl Genet, 103: 1254&ndash;126.</span></span></span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\">Cock PA, Antao T, Chang JT, Chapman BA, Cox CJ, Dalke A, Friedberg I, Hamelryck T, Kauff F, Wilczynski </span><span style=\"font-size:16px\">B and</span><span style=\"font-size:16px\"> de Hoon MJL (2009) Biopython: freely available Python tools for computational molecular biology and bioinformatics.&nbsp;<em>Bioinformatics</em>, 25, 1422-1423.</span></li>\r\n	<li style=\"text-align:justify\"><span style=\"font-size:16px\">Edgar, Robert C. (2004), MUSCLE: multiple sequence alignment with high accuracy and high throughput,&nbsp;<em>Nucleic Acids Research</em>&nbsp;<strong>32</strong>(5), 1792-97.</span></li>\r\n</ul>\r\n\r\n<p style=\"margin-left:0cm; margin-right:0cm; text-align:justify\">&nbsp;</p>\r\n',
        '', 'Theoretical Introduction to the Calculation', 1);
INSERT INTO `pages` (`id`, `name`, `slug`, `seo_desc`, `seo_keywords`, `text`, `desc`, `seo_title`, `is_active`)
VALUES (13, 'Privacy policy', '/privacy-policy', 'Description of privacy policy in out app gene-calc.pl',
        'privacy policy',
        '<h1>Privacy Policy</h1>\r\n\r\n<p>Effective date: October 15, 2018</p>\r\n\r\n<p>Gene Calc (&quot;us&quot;, &quot;we&quot;, or &quot;our&quot;) operates the https://gene-calc.pl/ website (the &quot;Service&quot;).</p>\r\n\r\n<p>This page informs you of our policies regarding the collection, use, and disclosure of personal data when you use our Service and the choices you have associated with that data. Our Privacy Policy for Gene Calc is managed through <a href=\"https://www.freeprivacypolicy.com/free-privacy-policy-generator.php\">Free Privacy Policy</a>.</p>\r\n\r\n<p>We use your data to provide and improve the Service. By using the Service, you agree to the collection and use of information in accordance with this policy. Unless otherwise defined in this Privacy Policy, terms used in this Privacy Policy have the same meanings as in our Terms and Conditions, accessible from https://gene-calc.pl/</p>\r\n\r\n<h2>Information Collection And Use</h2>\r\n\r\n<p>We collect several different types of information for various purposes to provide and improve our Service to you.</p>\r\n\r\n<h3>Types of Data Collected</h3>\r\n\r\n<h4>Personal Data</h4>\r\n\r\n<p>While using our Service, we may ask you to provide us with certain personally identifiable information that can be used to contact or identify you (&quot;Personal Data&quot;). Personally identifiable information may include, but is not limited to:</p>\r\n\r\n<ul>\r\n	<li>Email address</li>\r\n	<li>Cookies and Usage Data</li>\r\n</ul>\r\n\r\n<h4>Usage Data</h4>\r\n\r\n<p>We may also collect information how the Service is accessed and used (&quot;Usage Data&quot;). This Usage Data may include information such as your computer&#39;s Internet Protocol address (e.g. IP address), browser type, browser version, the pages of our Service that you visit, the time and date of your visit, the time spent on those pages, unique device identifiers and other diagnostic data.</p>\r\n\r\n<h4>Tracking &amp; Cookies Data</h4>\r\n\r\n<p>We use cookies and similar tracking technologies to track the activity on our Service and hold certain information.</p>\r\n\r\n<p>Cookies are files with small amount of data which may include an anonymous unique identifier. Cookies are sent to your browser from a website and stored on your device. Tracking technologies also used are beacons, tags, and scripts to collect and track information and to improve and analyze our Service.</p>\r\n\r\n<p>You can instruct your browser to refuse all cookies or to indicate when a cookie is being sent. However, if you do not accept cookies, you may not be able to use some portions of our Service.</p>\r\n\r\n<p>Examples of Cookies we use:</p>\r\n\r\n<ul>\r\n	<li><strong>Session Cookies.</strong> We use Session Cookies to operate our Service.</li>\r\n	<li><strong>Preference Cookies.</strong> We use Preference Cookies to remember your preferences and various settings.</li>\r\n	<li><strong>Security Cookies.</strong> We use Security Cookies for security purposes.</li>\r\n</ul>\r\n\r\n<h2>Use of Data</h2>\r\n\r\n<p>Gene Calc uses the collected data for various purposes:</p>\r\n\r\n<ul>\r\n	<li>To provide and maintain the Service</li>\r\n	<li>To notify you about changes to our Service</li>\r\n	<li>To allow you to participate in interactive features of our Service when you choose to do so</li>\r\n	<li>To provide customer care and support</li>\r\n	<li>To provide analysis or valuable information so that we can improve the Service</li>\r\n	<li>To monitor the usage of the Service</li>\r\n	<li>To detect, prevent and address technical issues</li>\r\n</ul>\r\n\r\n<h2>Transfer Of Data</h2>\r\n\r\n<p>Your information, including Personal Data, may be transferred to &mdash; and maintained on &mdash; computers located outside of your state, province, country or other governmental jurisdiction where the data protection laws may differ than those from your jurisdiction.</p>\r\n\r\n<p>If you are located outside Poland and choose to provide information to us, please note that we transfer the data, including Personal Data, to Poland and process it there.</p>\r\n\r\n<p>Your consent to this Privacy Policy followed by your submission of such information represents your agreement to that transfer.</p>\r\n\r\n<p>Gene Calc will take all steps reasonably necessary to ensure that your data is treated securely and in accordance with this Privacy Policy and no transfer of your Personal Data will take place to an organization or a country unless there are adequate controls in place including the security of your data and other personal information.</p>\r\n\r\n<h2>Disclosure Of Data</h2>\r\n\r\n<h3>Legal Requirements</h3>\r\n\r\n<p>Gene Calc may disclose your Personal Data in the good faith belief that such action is necessary to:</p>\r\n\r\n<ul>\r\n	<li>To comply with a legal obligation</li>\r\n	<li>To protect and defend the rights or property of Gene Calc</li>\r\n	<li>To prevent or investigate possible wrongdoing in connection with the Service</li>\r\n	<li>To protect the personal safety of users of the Service or the public</li>\r\n	<li>To protect against legal liability</li>\r\n</ul>\r\n\r\n<h2>Security Of Data</h2>\r\n\r\n<p>The security of your data is important to us, but remember that no method of transmission over the Internet, or method of electronic storage is 100% secure. While we strive to use commercially acceptable means to protect your Personal Data, we cannot guarantee its absolute security.</p>\r\n\r\n<h2>Service Providers</h2>\r\n\r\n<p>We may employ third party companies and individuals to facilitate our Service (&quot;Service Providers&quot;), to provide the Service on our behalf, to perform Service-related services or to assist us in analyzing how our Service is used.</p>\r\n\r\n<p>These third parties have access to your Personal Data only to perform these tasks on our behalf and are obligated not to disclose or use it for any other purpose.</p>\r\n\r\n<h3>Analytics</h3>\r\n\r\n<p>We may use third-party Service Providers to monitor and analyze the use of our Service.</p>\r\n\r\n<ul>\r\n	<li>\r\n	<p><strong>Google Analytics</strong></p>\r\n\r\n	<p>Google Analytics is a web analytics service offered by Google that tracks and reports website traffic. Google uses the data collected to track and monitor the use of our Service. This data is shared with other Google services. Google may use the collected data to contextualize and personalize the ads of its own advertising network.</p>\r\n\r\n	<p>You can opt-out of having made your activity on the Service available to Google Analytics by installing the Google Analytics opt-out browser add-on. The add-on prevents the Google Analytics JavaScript (ga.js, analytics.js, and dc.js) from sharing information with Google Analytics about visits activity.</p>\r\n\r\n	<p>For more information on the privacy practices of Google, please visit the Google Privacy &amp; Terms web page: <a href=\"https://policies.google.com/privacy?hl=en\">https://policies.google.com/privacy?hl=en</a></p>\r\n	</li>\r\n</ul>\r\n\r\n<h2>Links To Other Sites</h2>\r\n\r\n<p>Our Service may contain links to other sites that are not operated by us. If you click on a third party link, you will be directed to that third party&#39;s site. We strongly advise you to review the Privacy Policy of every site you visit.</p>\r\n\r\n<p>We have no control over and assume no responsibility for the content, privacy policies or practices of any third party sites or services.</p>\r\n\r\n<h2>Children&#39;s Privacy</h2>\r\n\r\n<p>Our Service does not address anyone under the age of 18 (&quot;Children&quot;).</p>\r\n\r\n<p>We do not knowingly collect personally identifiable information from anyone under the age of 18. If you are a parent or guardian and you are aware that your Children has provided us with Personal Data, please contact us. If we become aware that we have collected Personal Data from children without verification of parental consent, we take steps to remove that information from our servers.</p>\r\n\r\n<h2>Changes To This Privacy Policy</h2>\r\n\r\n<p>We may update our Privacy Policy from time to time. We will notify you of any changes by posting the new Privacy Policy on this page.</p>\r\n\r\n<p>We will let you know via email and/or a prominent notice on our Service, prior to the change becoming effective and update the &quot;effective date&quot; at the top of this Privacy Policy.</p>\r\n\r\n<p>You are advised to review this Privacy Policy periodically for any changes. Changes to this Privacy Policy are effective when they are posted on this page.</p>\r\n\r\n<h2>Contact Us</h2>\r\n\r\n<p>If you have any questions about this Privacy Policy, please contact us:</p>\r\n\r\n<ul>\r\n	<li>By email: contact@gene-calc.pl</li>\r\n	<li>By visiting this page on our website: https://gene-calc.pl/contact</li>\r\n</ul>\r\n',
        '<p><strong>of gene-calc.pl</strong></p>\r\n', 'Privacy policy of gene-calc.pl', 0);
INSERT INTO `pages` (`id`, `name`, `slug`, `seo_desc`, `seo_keywords`, `text`, `desc`, `seo_title`, `is_active`)
VALUES (14, 'APMC', '/apmc', 'SEO desc',
        'seo keywords', '<p>APMC - text</p>',
        '<p>APMC - automatic predictive model constructor</p>', 'seo title', 1);

INSERT INTO `customers` (`first_name`, `last_name`, `profile_pic`, `login`, `email`, `password`, `created_at`,
                         `is_superuser`)
VALUES ('test', 'test', 'profile.svg', 'test123', 'test123@test.com',
        'sha256$OUUSNCtW$51c276a09fac6fedc7aae1c6c1f669c3a2c5eab554341c0b48ebc42691f113b1', '2019-08-02 10:17:17', 1);
